### PR TITLE
`assertMinValueOverflow` should throw same type of exception as Spark

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>
Related to #5196 

## Changes in the PR
1. Throw same type of exception as Spark in `assertMinValueOverflow`. For 
For Spark 3.1 and 3.2, `java.lang.ArithmeticException` is thrown.
For Spark 3.3 and later versions, `SparkArithmeticException` is thrown: 
https://github.com/apache/spark/blob/branch-3.3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/MathUtils.scala#L71-L76
https://github.com/apache/spark/blob/branch-3.3/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala#L492-L503
2. Update `test_unary_minus_ansi_overflow`, `test_unary_minus_ansi_overflow_day_time_interval`, `test_abs_ansi_overflow_day_time_interval` and `test_abs_ansi_overflow` to check the type of exceptions.

### CODE REVIEW

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
Met an odd error: https://github.com/NVIDIA/spark-rapids/runs/6914487613?check_suite_focus=true#step:2:36294, which is not related to this PR.

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
build

File: integration_tests/src/main/python/arithmetic_ops_test.py [hunk 1]
```diff
@@ -378,7 +384,8 @@ def test_abs_ansi_overflow(data_type, value):
     assert_gpu_and_cpu_error(
             df_fun=lambda spark: _get_overflow_df(spark, [value], data_type, 'abs(a)').collect(),
             conf=ansi_enabled_conf,
-            error_message='ArithmeticException')
```
Comment (revans2) [inline integration_tests/src/main/python/arithmetic_ops_test.py:381]: Do we need/want a similar "We don't check the error message" comment on all of these?

**Review by revans2 — APPROVED** [review]
Just a nit about comments. What would it take to actually output an error message that matches what Spark does? Performance is not really that big of a deal because we already know that we are going to fail so doing some more computation at that point is probably not that big of a deal.

File: integration_tests/src/main/python/arithmetic_ops_test.py (hunk 1 above)
Comment (HaoYang670) [reply integration_tests/src/main/python/arithmetic_ops_test.py:381]: Updated!

**Comment by HaoYang670** [comment]
build

**Comment by HaoYang670** [comment]
> What would it take to actually output an error message that matches what Spark does?

No much work I guess. But personally, I prefer the error message that our plugin gives, because it tells users which operation is error.

**Review by revans2 — APPROVED** [review]

**Pull request merged**